kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/source/config/kcqlprops/ReadTextMode.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2017-2023 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.aws.s3.source.config.kcqlprops
+
+import enumeratum._
+
+sealed trait ReadTextModeEntry extends EnumEntry
+
+object ReadTextModeEnum extends Enum[ReadTextModeEntry] {
+
+  override val values = findValues
+
+  case object Regex       extends ReadTextModeEntry
```

Comment:
What happens with the default mode where it reads the data as it is today.

Reply:
It is an optional, so no `ReadTextMode` is set as the default behaviour.